mqt/qcec/parameterized.py

```diff
@@ -0,0 +1,174 @@
+    param_map = {}
+    for p in params:
+        param_map[p] = np.random.rand() * 2 * np.pi
+
+    circ1_inst = circ1.bind_parameters(param_map)
+    circ2_inst = circ2.bind_parameters(param_map)
+
+    return check_instantiated(circ1_inst, circ2_inst, configuration, **kwargs)
+
+
+def __parse_args(configuration: Configuration | None = None, **kwargs: Any) -> tuple[int, float]:
+    n_checks = 0
+    tol = 1e-12
+    if kwargs:
+        if kwargs.get("additional_instantiations"):
+            n_checks = kwargs["additional_instantiations"]
+        if kwargs.get("parameterized_tolerance"):
+            tol = kwargs["parameterized_tolerance"]
+    elif configuration is not None:
+        n_checks = configuration.parameterized.additional_instantiations
+        tol = configuration.parameterized.parameterized_tolerance
+    return n_checks, tol
+
+
+def check_parameterized(
+    circ1: QuantumCircuit, circ2: QuantumCircuit, configuration: Configuration | None = None, **kwargs: Any
+) -> EquivalenceCheckingManager.EquivalenceCheckingManager.Results:
+    """Equivalence checking flow for parameterized circuit."""
+    res = check_parameterized_zx(circ1, circ2, configuration, **kwargs)
+
+    if res.considered_equivalent():
+        return res
+
+    n_checks, tol = __parse_args(configuration, **kwargs)
+
+    parameters, A, offsets = extract_params(circ1, circ2)
+
+    def instantiate_params(
+        qc1: QuantumCircuit, qc2: QuantumCircuit, b: np.array
+    ) -> tuple[QuantumCircuit, QuantumCircuit]:
+        A_pinv = np.linalg.pinv(A)
+        x = np.dot(A_pinv, b)
+        param_map = {param: x[i] for i, param in enumerate(parameters)}
+        qc1_bound = qc1.bind_parameters(param_map)
+        qc2_bound = qc2.bind_parameters(param_map)
+
+        def round_zero_params(qc: QuantumCircuit) -> QuantumCircuit:
+            for instr in qc.data:
+                params = instr[0].params
+                instr[0].params = [float(x) for x in params]
+                instr[0].params = [0 if np.abs(x) < tol else x for x in instr[0].params]
+            return qc
+
+        qc1_bound = round_zero_params(qc1_bound)
+        qc2_bound = round_zero_params(qc2_bound)
+        return (qc1_bound, qc2_bound)
+
+    def instantiate_params_zero(qc1: QuantumCircuit, qc2: QuantumCircuit) -> tuple[QuantumCircuit, QuantumCircuit]:
+        return instantiate_params(qc1, qc2, offsets)
+
+    def instantiate_params_phases(qc1: QuantumCircuit, qc2: QuantumCircuit) -> tuple[QuantumCircuit, QuantumCircuit]:
+        phases = [0, np.pi, np.pi / 2, -np.pi / 2, np.pi / 4, -np.pi / 4]
+        b = np.random.choice(phases, size=len(offsets)) + offsets
+        return instantiate_params(qc1, qc2, b)
+
+    circ1_inst, circ2_inst = instantiate_params_zero(circ1, circ2)
+    res = check_instantiated(circ1_inst, circ2_inst, configuration, **kwargs)
+    if res.equivalence == "not_equivalent":
+        return res
+
+    for _i in range(1, n_checks - 1):
```

Comment:
shouldn't `n_checks` happen here? I thought that parameter configures the number of additional instantiations. (maybe a better name should be used here). Then, this should be `range(n_checks)`

Reply:
That is a leftover from how it was handled previously. Changed it back.